### ISSUE TITLE
import-rna on 3'-end RNA sequencing

### ISSUE THREAD

**Issue by laopp** [etal/cnvkit#352]
Hi I'm trying to run import-rna with raw read counts. The results are very different from that by targeted DNA seq. The samples for RNA seq vs. DNA seq are 99% same so they are supposed to generate similar results. My question is that our samples are FFPE (formalin-fixed) archival tissues and  sequenced by the 3'-end sequencing methods rather than the traditional RNA seq. The method generates a directional sequencing library comprised primarily of ~200 bp cDNA fragments with a poly-A tail, in which sequencing will proceed directionally toward the poly-A tail. Do you think that import-rna will work on the 3’-seq data? Thanks!

**Comment by etal** [comment]
Under the best of conditions RNA-based copy number inference is only accurate for large-scale alterations, and these are not the best of conditions. But if you were able to run RSEM to quantify this RNA-seq data then it should work to some extent with `import-rna`. Do the RNA and DNA-based calls at least agree at the chromosome arm level? Or are the RNA results pure noise?

**Comment by laopp** [comment]
Here are the heatmaps from DNA seq data and RNA data. 
![image](https://user-images.githubusercontent.com/39168066/39953612-8a25c53c-5564-11e8-8ed3-a1bf11c792ce.png)
![image](https://user-images.githubusercontent.com/39168066/39953613-8f3f45de-5564-11e8-9548-df65dafc71a0.png)
The samples are in the same order except that normal is included in the RNA results at the very bottom. You can see that there are some degree of consistency for 1q and 8, but wonder why RNA data is so blue in general. Also we would expect normal to be close to zero, but it turns out to be very negative. Any thoughts?

**Comment by laopp** [comment]
Yes the centering on RNA samples is very helpful. The gain at 1q and 8p does show up on the heatmap. But I wonder why normal sample does not show 0 across all chromosomes, are the samples normalized against the normal samples? 

I also did centering on the DNA samples, and I don't see much improvement. some light blue gets even lighter after the centering.

![image](https://user-images.githubusercontent.com/39168066/40009523-265a8066-5757-11e8-9768-2f7b5f808b56.png)

![image](https://user-images.githubusercontent.com/39168066/40009550-3494b53e-5757-11e8-9f14-ee24d0de335d.png)

**Comment by etal** [comment]
Currently the `-n` option in `import-rna` doesn't do anything -- it's an unimplemented feature that I probably should have hidden before pushing out the last release. (Or finished implementing it.)

If the majority of your RNA samples have the same copy number alteration, then that alteration will be treated as the baseline and your normal/control sample will show a spurious alteration in the opposite direction. Is that what happened here?

**Comment by etal** [comment]
The -n/--normal option is in place now (#362), if you'd like to try it out.

**Comment by laopp** [comment]
Hi Eric,

Thanks for letting me know. Will definitely give it a try!

Thanks,

Peipei

**Comment by laopp** [comment]
Hi Eric, I tried the --normal option in import-rna. It seems that the blue/red are getting even more lighter after the normalization. Also, should I expect that the normal sample has a near 0 baseline on the heatmap? Obviously I didn't see that. Thanks!
![image](https://user-images.githubusercontent.com/39168066/41746944-0c66d8f8-7561-11e8-97ff-470ac02ea8a4.png)

**Comment by sxv** [comment]
Hi Eric,

Thanks for implementing this. I'm also wondering about the issue above. I wouldn't expect to see such gains or losses in the normal sample(s). Any thoughts on what could be causing this result?

Sujay

**Comment by etal** [comment]
Not really sure, this part of the code is less tested and could use another set of eyes on it.

**Comment by etal** [comment]
**Verify-FIRST investigation outcome: not reproducible on current master; design weakness documented.**

Attempted to reproduce the reported symptom (normal sample heatmap not centered; dynamic-range compression) across synthetic cohorts spanning {1, 3, 4} normal samples × {clean / contaminated / cohort-wide-altered} configurations, at both the `normalize_read_depths` unit level and the full `import-rna` pipeline. In every case, the normal sample's `.cnr` came out at `log2 ≈ 0` and tumor gains/losses recovered at the expected log2 values.

**Why a single normal looks "flat zero" by construction.** When `--normal X` is used with a single normal sample, the per-gene "median of normals" used as the anchor reduces to that sample's own column, so the anchoring step becomes a self-divide and the normal's `.cnr` is approximately zero at every gene. This is mathematically self-consistent — `log2(normal/normal) = 0` — but produces no QC signal on the normal itself, which is a likely interpretation of the original observation.

**Dynamic-range-compression hypothesis.** The "blue/red getting lighter" symptom is consistent with a cohort in which the normals share alterations with the tumors (stromal contamination in adjacent-tissue normals, field effects in matched-tissue controls). When this happens, the normal-anchored divide cancels real tumor signal and reduces the dynamic range of the resulting calls. Pure tissue-matched controls from independent donors are preferred when available.

**Recommendations when using `--normal`:**

- Use at least three process-matched normals if at all possible. With fewer than three, the per-gene normal median is a high-variance estimator; with exactly one, the normal's own output is tautological.
- Verify the normals are free of the alterations being studied.
- With a single normal, expect the normal's own `.cnr` to be ≈ 0 everywhere by construction — this is the design behaviour, not a QC artifact.

A `logging.warning` is now emitted by `import-rna` when `0 < n_normals < 3` (PR #1098), and the design behaviour, the small-cohort caveats, and the contaminated-normals failure mode are documented in `doc/rna.rst` under "Using ``--normal`` to anchor against control samples". A deeper redesign of the `--normal` anchoring (DESeq2-style size factors estimated from normals only, plus leave-one-out anchoring so multi-normal cohorts retain QC signal in the normals' own `.cnr` files) is on the roadmap.

Closing as documented. If the symptom persists on current master with a configuration that meets the recommendations above, please open a new issue with a minimal reproducer (counts files + gene resource snippet, or a synthetic generator).
